ext/AvizExt/viz/sensitivity.jl

```diff
@@ -244,6 +273,199 @@ function ADRIA.viz.rsa(rs::ResultSet, si::NamedDimsArray, factors::Vector{String
 end
 
 
+"""
+    ADRIA.viz.convergence(Si_conv::NamedDimsArray, factors::Vector{Symbol}; series_opts::Dict=Dict(),
+        axis_opts::Dict=Dict())
+    ADRIA.viz.convergence(Si_conv::NamedDimsArray, factors::Vector{Symbol}, plot_overlay::Bool; series_opts::Dict=Dict(),
+        axis_opts::Dict=Dict())
+    ADRIA.viz.convergence!(f::Figure, Si_conv::NamedDimsArray, factors::Vector{Symbol}; series_opts::Dict=Dict(), 
+        axis_opts::Dict=Dict())
+    ADRIA.viz.convergence!(g::GridPosition, Si_conv::NamedDimsArray, factors::Vector{Symbol}, plot_overlay::Bool; 
+        axis_opts::Dict=Dict())
+
+Plot sensitivty metric for increasing number of scenarios to illustrate convergence.
+
+# Arguments
+- `Si_conv` : Produced using ADRIA.analysis.convergence() 
+- `factors` : Factors/model components to plot.
+- `plot_overlay` : If included band plots are plotted, if not included a heatmap is plotted. 
+    If true, for each factor the plots are overlayed. If false, each factor is plotted separately in a grid.
+- `opts` : Additional figure customization options
+- `fig_opts` : Additional options to pass to adjust Figure creation
+  See: https://docs.makie.org/v0.19/api/index.html#Figure
+- `series_opts` : Additional options to pass to adjust Series attributes
+  See: https://docs.makie.org/v0.19/api/index.html#series!
+- `axis_opts` : Additional options to pass to adjust Axis attributes
+  See: https://docs.makie.org/v0.19/api/index.html#Axis
+
+# Returns
+GLMakie figure
+"""
+function ADRIA.viz.convergence!(
+    g::GridPosition,
+    Si_conv::NamedDimsArray,
+    factors::Vector{Symbol},
```

Comment:
How does this differentiate between factors and groups of factors related to a specific component?
I don't see how the usage in the docs works.

I'm not sure if its clear to you, but any method specific option should be in `opts`, not as a function argument.
Specifically, `plot_overlay` should be passed in as `opts=Dict(:plot_overlay=>true)`

I think `Dict(:viz_type=>:heatmap)` and `Dict(:viz_type=>:series)` is more explicit.

I would use a barrier function and separate out the functionality according to whether its a heatmap or a series plot.

```
function _series_convergence()
    ...
end

function _heatmap_convergence()
    ...
end

function ADRIA.viz.convergence!(
    ....
    opts=Dict(:viz_type=>:series)
)
    viz_type = get(opts, :viz_type, :series)
    
    try
        func = eval(Symbol("_$(viz_type)_convergence"))
    catch err
        if !(err isa UndefVarError)
            rethrow(err)
        end

        error("Convergence plot type $(viz_type) is not supported")
    end

    return func(pass_in_relevant_args_here)
end
```

Adding support for other kinds of analyses would then just be a matter of adding more private functions with the pattern `_*_convergence()`, where `*` is the name of the approach.

Reply:
Hey @ConnectedSystems, thanks for the suggestion. I've made changes to implement this but am having trouble with the scope of eval(). It throws an undefined error due to function inputs (g, factors etc.) not being in scope. I tried to avoid this by switching to `func = getfield(@__MODULE__, Symbol("_$(viz_type)_convergence"))`, but then it evaluates the function fine (as in inputs are in scope) but throws an error on return because func is undefined outside of the `try-catch` block. It works if I do `global func = getfield(@__MODULE__, Symbol("_$(viz_type)_convergence"))` but this is a hacky fix. Any idea how to solve this?

Reply:
Thanks, I think I'll switch to `if/elseif/else` for now because I can't think of another plot type case. If we add one in future I can always change it.